### DISCUSSION TITLE
Persist user home directory?

### DISCUSSION THREAD

**Discussion by baodrate** [hassio-addons/addon-ssh#641] [Ideas]
Can the user home directory be set to the [new public addon config directory](https://developers.home-assistant.io/blog/2023/11/06/public-addon-config/)? This would let the user persist their home directory (`.zshrc`/bindings/etc.)

Meaning this line should be removed/moved behind a `[ ! -f "/home/${username}/.profile" ]` check as to not override the user's `.profile`:

https://github.com/hassio-addons/addon-ssh/blob/091777cc91ef7f41e555af15c5f078283a9f656b/ssh/rootfs/etc/s6-overlay/s6-rc.d/init-ssh/run#L109-L111

**Comment by frenck** [comment]
This add-on cannot, as it uses the old binding as well.

As a matter of fact, there is no need to either, as it can be access through the add-on itself already :)

../Frenck

**Reply by baodrate** [reply]
> it uses the old binding as well

Can you elaborate? I see these lines:

https://github.com/hassio-addons/addon-ssh/blob/091777cc91ef7f41e555af15c5f078283a9f656b/ssh/rootfs/etc/s6-overlay/s6-rc.d/init-user/run#L22C1-L26

Is the `/config` required by the addon itself or is it purely convenience for the user

> there is no need to either, as it can be access through the add-on itself already

What do you mean by this? The user's home directory (`/home/hassio` by default) is not persisted. Neither is `/root`. Any changes to those directories (e.g. `.zshrc`/`.zsh_history`/`.config`/etc.) are wiped when the addon restarts
